PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SurveyPopUpDialog.java

```diff
@@ -0,0 +1,224 @@
+package com.microsoft.azuretools.ijidea.ui;
+
+import com.intellij.ide.ui.LafManager;
+import com.intellij.ide.ui.LafManagerListener;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.impl.ProjectManagerImpl;
+import com.intellij.openapi.util.IconLoader;
+import com.intellij.openapi.wm.impl.WindowManagerImpl;
+import com.intellij.ui.HyperlinkLabel;
+import com.microsoft.intellij.helpers.CustomerSurveyHelper;
+import com.microsoft.intellij.util.PluginUtil;
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.*;
+import javax.swing.event.MouseInputAdapter;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseMotionAdapter;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+
+public class SurveyPopUpDialog extends JDialog {
+
+    private static final int DISPOSE_TIME = 10;
+    private static final String LABEL_DO_NOT_SHOW_AGAIN = "Don't show again.";
+    private static final String LABEL_PROMPT = "<html>" +
+            "<h3 align=\"center\">Enjoy Java Toolkits?</h3>" +
+            "<p align=\"center\" style=\"line-height: 0%\"> Your feedback is important, </p>" +
+            "<p align=\"center\" style=\"line-height: 0%\"> take a minute to fill out our survey.</p>" +
+            "</html>";
+
+    private JPanel contentPane;
+    private JButton giveFeedbackButton;
+    private JButton notNowButton;
+    private HyperlinkLabel lblDoNotShowAgain;
+    private JLabel lblMessage;
+    private JLabel lblAzureIcon;
+    private JLabel lblClose;
+
+    private CustomerSurveyHelper customerSurveyHelper;
+
+    private Point dragPosition;
+    private Timer disposeTimer;
+    private LafManagerListener themeListener;
+    private Color buttonOnHoverColor = Color.WHITE;
+
+    public SurveyPopUpDialog(CustomerSurveyHelper customerSurveyHelper, Project project) {
+        super();
+
+        this.customerSurveyHelper = customerSurveyHelper;
+        this.disposeTimer = new Timer(1000* DISPOSE_TIME, (e)->this.putOff());
+        this.themeListener = lafManager -> renderUiByTheme(lafManager.getCurrentLookAndFeel());
+
+        this.setAlwaysOnTop(true);
+        this.setSize(250, 250);
+        this.setContentPane(contentPane);
+        this.setUndecorated(true);
+        this.getRootPane().setBorder(BorderFactory.createLineBorder(Color.GRAY));
+        this.setModal(false);
+        this.setLocationRelativeToIDE(project);
+        this.setDisposeTimer();
+
+        lblMessage.setText(LABEL_PROMPT);
```

Comment:
The form related components are not safe to access in constructor.  It has to be put into `protected void createUIComponents()` with `Custom Create` enabled.

Reply:
Fixed~